client/consensus/slots/src/lib.rs

```diff
@@ -130,11 +133,11 @@ pub trait SimpleSlotWorker<B: BlockT> {
 
 	/// Tries to claim the given slot, returning an object with claim data if successful.
 	fn claim_slot(
```

Comment:
Why not make it just async? We already use async_trait here

Reply:
Boilerplate that `async_trait` generates is not flexible enough in this case.

It generates return type that expects `Send` future by default:
```
error: future cannot be sent between threads safely
   --> client/consensus/aura/src/lib.rs:366:27
    |
366 |       ) -> Option<Self::Claim> {
    |  ______________________________^
367 | |         let expected_author = slot_author::<P>(slot, &epoch_data);
368 | |         let maybe_claim = expected_author.and_then(|p| {
369 | |             if SyncCryptoStore::has_keys(
...   |
379 | |         maybe_claim
380 | |     }
    | |_____^ future created by async block is not `Send`
    |
note: captured value is not `Send` because `&` references cannot be sent unless their referent is `Sync`
   --> client/consensus/aura/src/lib.rs:362:8
    |
362 |         &mut self,
    |              ^^^^ has type `&mut AuraWorker<C, E, I, P, SO, L, BS>` which is not `Send`, because `AuraWorker<C, E, I, P, SO, L, BS>` is not `Sync`
    = note: required for the cast to the object type `dyn futures::Future<Output = std::option::Option<<P as sp_application_crypto::Pair>::Public>> + std::marker::Send`
help: consider further restricting this bound
    |
321 |     E: Environment<B, Error = Error> + std::marker::Send,
    |                                      +++++++++++++++++++

error: could not compile `sc-consensus-aura` due to previous error
```

Changing attribute to `#[async_trait::async_trait(?Send)]` flips the behavior and causes cascading errors in other places that worked before as macro can either generate `Send` or `!Send` futures, but not both at the same time.

The approach chosen actually mimics such of `fn proposer()` in this same trait. Async story in Rust is still tricky at times.